source/NationalInstruments/jquery.flot.highlights.js

```diff
@@ -0,0 +1,210 @@
+                        let pointIndex = highlightIndexes[i];
+                        if (this._options.highlightPoints) {
+                            this._drawPointHighlight(series[j], plot, ctx, pointIndex);
+                        }
+
+                        if (this._options.highlightBars) {
+                            this._drawBarHighlight(series[j], [series[j].datapoints.points[pointIndex * ps], series[j].datapoints.points[pointIndex * ps + 1]], ctx);
+                        }
+                    }
+                }
+
+                let highlightRange = this._options.selectedRange[j];
+                if (highlightRange) {
+                    for (let i = highlightRange[0]; i < highlightRange[1]; i++) {
+                        if (this._options.highlightPoints) {
+                            this._drawPointHighlight(series[j], plot, ctx, i);
+                        }
+
+                        if (this._options.highlightBars) {
+                            this._drawBarHighlight(series[j], [series[j].datapoints.points[i * ps], series[j].datapoints.points[i * ps + 1]], ctx);
+                        }
+                    }
+                }
+
+                if (this._options.highlightLines && highlightIndexes.length > 0) {
+                    this._drawLineHighlight(series[j], plot, ctx);
+                }
+            }
+
+            ctx.restore();
+        }
+
+        _drawPointHighlight(series, plot, ctx, pointIndex) {
+            const ps = series.datapoints.pointsize;
+            let x = series.datapoints.points[pointIndex * ps];
+            let y = series.datapoints.points[pointIndex * ps + 1];
+            const axisx = series.xaxis;
+            const axisy = series.yaxis;
+            const highlightColor = (typeof this._options.highlightColor === "string") ? this._options.highlightColor : $.color.parse(series.color).scale('a', 0.5).toString();
+
+            if (x < axisx.min || x > axisx.max || y < axisy.min || y > axisy.max) {
+                return;
+            }
+
+            const pointRadius = series.points.radius + series.points.lineWidth / 2;
+            ctx.lineWidth = pointRadius;
+            ctx.strokeStyle = highlightColor;
+            ctx.fillStyle = highlightColor;
+            ctx.shadowColor = 'black';
+            ctx.shadowBlur = 3;
+            ctx.shadowOffsetY = 1;
+            let radius = 1.5 * pointRadius;
+            x = axisx.p2c(x);
+            y = axisy.p2c(y);
+
+            ctx.beginPath();
+            const symbol = series.points.symbol;
+            if (symbol === 'circle') {
+                ctx.arc(x, y, radius, 0, 2 * Math.PI, false);
+            } else if (typeof symbol === 'string' && plot.drawSymbol && plot.drawSymbol[symbol]) {
+                plot.drawSymbol[symbol](ctx, x, y, radius, false);
+            }
+
+            ctx.closePath();
+            ctx.fill();
+        }
+
+        _drawLineHighlight(series, plot, ctx) {
```

Comment:
This seems to be highlighting the entire line. Why aren't we just highlighting the user-specified sections?

Reply:
This is intentional - you don't have to know anything about the line other than its series index to draw the entire line.

Reply:
Ranges are now applied to lines